content/player/blockchain-integration/buying-nfts.md

```diff
@@ -0,0 +1,237 @@
+
+When you are buying items from a collection this way, you are actually minting a new NFT that can then be commercialized with the Polygon Marketplace or Bids smart contract.
+
+If a collection is on sale, any user with the sufficient amount of MANA can buy an item from it as long as there is stock left to be minted.
+
+The CollectionStore smart contract exposes the `buy` function to do so.
+
+```sol
+/// @param _itemsToBuy A list of items from different collections that the caller want to buy.
+function buy(ItemToBuy[] memory _itemsToBuy)
+```
+
+```sol
+struct ItemToBuy {
+  // The address of the collection the items in the ids array belong to.
+  IERC721CollectionV2 collection;
+  // The id of the particular items in the collection that the caller want to buy.
+  uint256[] ids;
+  // The price that the buyer wants to pay for the item in the same index.
+  // Must be equal to the price set by the creator.
+  uint256[] prices;
+  // The addresses that will receive the minted items in the same index.
+  address[] beneficiaries;
+}
+```
+
+To make this easier to understand. Imagine that I want to buy item "Spooky Hat" with item id 0 from the collection called "Halloween Madness".
+
+The hat has been determined by the creator to have a price of 20 MANA.
+
+To buy the item I would have to call the `buy` function with the following data:
+
+```js
+buy([
+  [
+    "0xHalloweenMadnessAddress",
+    ["0"], // The item id of the Spooky Hat, other items in the collection will have different incrementing ids.
+    ["20000000000000000000"], // Which is the equivalent in wei to 20
+    ["My own address"],
+  ],
+]);
+```
+
+When the transaction to buy the item is successful. 20 MANA will be transferred from my address to the creator (or a beneficiary defined by the creator), the hat will be minted as an NFT and transferred to me.
+
+## Testnet Smart Contracts
+
+These are the smart contracts that can be found on the respective Ethereum and Polygon testnets.
+
+|Contract|Network|Address|
+|:-|:-|:-|
+|Marketplace|Goerli|[0x5d01fbD3E22892be40F69bdAE7Ad921C8cdA2085](https://goerli.etherscan.io/address/0x5d01fbD3E22892be40F69bdAE7Ad921C8cdA2085)|
+|Bids|Goerli|[0xd7dC1C183B8fFaED6b7f30fFC616Ff81B66812e5](https://goerli.etherscan.io/address/0xd7dC1C183B8fFaED6b7f30fFC616Ff81B66812e5)|
+|Marketplace|Mumbai|[0x5A467398dfa9d5C663a656423A2D055f538198A4](https://mumbai.polygonscan.com/address/0x5A467398dfa9d5C663a656423A2D055f538198A4)|
+|Bids|Mumbai|[0x78Dd92c8941dBC7BE54E2a9390D58aD28AD97afD](https://mumbai.polygonscan.com/address/0x78Dd92c8941dBC7BE54E2a9390D58aD28AD97afD)|
+|CollectionStore|Mumbai|[0x6ddF1b1924DAD850AdBc1C02026535464Be06B0c](https://mumbai.polygonscan.com/address/0x6ddF1b1924DAD850AdBc1C02026535464Be06B0c)|
```

Comment:
I would move the contracts to the table at the top of the doc. Leave empty if the contract is not deployed in the chain